src/prognostic_variables.jl

```diff
@@ -92,7 +91,199 @@ function Base.zeros(::Type{PrognosticVariables{NF}},
                     nlev::Integer) where NF
 
     layers = [zeros(PrognosticVariablesLeapfrog{NF},model,lmax,mmax) for _ in 1:nlev]   # k layers
-    _lmax,_mmax = model isa BarotropicModel ? (-2,-1) : (lmax,mmax)   # pressure not needed for BarotropicModel
-    pres = zeros(SurfaceLeapfrog{NF},_lmax,_mmax)                     # 2 leapfrog time steps for pres
-    return PrognosticVariables(layers,pres,lmax,mmax,N_LEAPFROG,nlev)
+    pres = has(model, :pres) ? zeros(SurfaceLeapfrog{NF},lmax,mmax) : zeros(SurfaceLeapfrog{NF},-2,-1)  # 2 leapfrog time steps for pres       
+    return PrognosticVariables{NF,typeof(model)}(layers,pres,lmax,mmax,N_LEAPFROG,nlev)
+end
+
+has(progn::PrognosticVariables{NF,M}, var_name::Symbol) where {NF,M} = has(M, var_name)
+
+# SET_VAR FUNCTIONS TO ASSIGN NEW VALUES TO PrognosticVariables
+
+"""
+    set_var!(progn::PrognosticVariables{NF},        
+             varname::Symbol, 
+             var::Vector{<:LowerTriangularMatrix};
+             lf::Integer=1) where NF
+
+Sets the prognostic variable with the name `varname` in all layers at leapfrog index `lf` 
+with values given in `var` a vector with all information for all layers in spectral space.
+"""
+function set_var!(progn::PrognosticVariables{NF}, 
+                  varname::Symbol, 
+                  var::Vector{<:LowerTriangularMatrix};
+                  lf::Integer=1) where NF
+
+    @assert length(var) == length(progn.layers)
+    @assert has(progn, varname)
```

Comment:
Reminder to self that we should do something about error messages in general. I like the `@assert` here as they actually print what's wrong. However, often I ended up using `throw(BoundsError)` which is kind of clear if there's only one `@boundscheck` before a loop in a function but if there's several ones, you don't know which array sizes don't match. Maybe we can make them somehow more telling...

Reply:
I didn't know that before, but `@assert` let's you specify error messages: 
`@assert has(progn, varname) "No variable with this name found."`